src/Blockcore.Indexer.Core/Sync/SyncTasks/BlockPuller.cs

```diff
@@ -85,77 +89,147 @@ public override async Task<bool> OnExecute()
             return false;
          }
 
-         watch.Restart();
+         if (blockInfos.Count > 10000)
```

Comment:
this should become a config param or alternatively use the `config.MaxItemsInQueue` if it snot used anymore or maybe it can be used together